opentelemetry/proto/profiles/v1experimental/profiles.proto

```diff
@@ -120,42 +129,143 @@ message ResourceProfiles {
+  // for ease of understanding data migration, it is not intended that pprof:Profile and
+  // OpenTelemetry:Profile encoding be wire compatible.
+
+  // A description of the samples associated with each Sample.value.
+  // For a cpu profile this might be:
+  //   [["cpu","nanoseconds"]] or [["wall","seconds"]] or [["syscall","count"]]
+  // For a heap profile, this might be:
+  //   [["allocations","count"], ["space","bytes"]],
+  // If one of the values represents the number of events represented
+  // by the sample, by convention it should be at index 0 and use
+  // sample_type.unit == "count".
+  repeated ValueType sample_type = 1;
+  // The set of samples recorded in this profile.
+  repeated Sample sample = 2;
+  // Mapping from address ranges to the image/binary/library mapped
+  // into that address range.  mapping_table[0] will be the main binary.
+  repeated Mapping mapping_table = 3;
+  // Locations referenced by samples via location_indices.
+  repeated Location location_table = 4;
+  // Functions referenced by locations.
+  repeated Function function_table = 5;
+  // A common table for strings referenced by various messages.
+  // string_table[0] must always be "".
+  repeated string string_table = 6;
+  // frames with Function.function_name fully matching the following
+  // regexp will be dropped from the samples, along with their successors.
+  int32 drop_frames_strindex = 7;   // Index into string table.
+  // frames with Function.function_name fully matching the following
+  // regexp will be kept, even if it matches drop_frames.
+  int32 keep_frames_strindex = 8;  // Index into string table.
+
+  // The following pprof fields are informational, do not affect
+  // interpretation of results.
+
+  // Time of collection (UTC) represented as nanoseconds past the epoch.
+  int64 time_nanos = 9;
+  // Duration of the profile, if a duration makes sense.
+  int64 duration_nanos = 10;
+  // The kind of events between sampled occurrences.
+  // e.g [ "cpu","cycles" ] or [ "heap","bytes" ]
+  ValueType period_type = 11;
+  // The number of events between sampled occurrences.
+  int64 period = 12;
+  // Free-form text associated with the profile. The text is displayed as is
+  // to the user by the tools that read profiles (e.g. by pprof). This field
+  // should not be used to store any machine-readable information, it is only
+  // for human-friendly content. The profile must stay functional if this field
+  // is cleaned.
+  repeated int32 comment_strindices = 13; // Indices into string table.
+  // Index into the string table of the type of the preferred sample
+  // value. If unset, clients should default to the last sample value.
+  int32 default_sample_type_strindex = 14;
+
+  reserved 15; // doc_url field is not inherited from pprof.
+
+  // In the event that future pprof versions add additional fields, then
+  // future OpenTelementry versions MAY use these for corresponding purposes.
+  reserved 16,17,18,19;
```

Comment:
Yes, it may well be futile, but OTOH it's essentially zero cost to try. Whilst there are several other messages where we could do this, only a couple are ones where we already add additional fields without leaving a gap. The biggest mess is Sample, which has gaps for no reason and without using 'reserved' for them.
There is going to be at least one more PR for changes soon, I'm inclined to leave this as is for now and change it in that later one if needed, after a policy discussion with the SIG to get some global consistency on our approach. Unless you feel really strongly about it I'd prefer to get this PR merged without waiting on that, as the loss of change history is going to make re-syncing other changes a pain if it goes on too long.

Reply:
We can discuss this in a later PR, yes. My understanding was that we are going to remove fields that act as "the old way of doing things" (such as labels) and so at the physical representation level the OTel proto simply completely diverges from the pprof's proto. Plus we'd re-number and re-order the OTel's proto fields and so the numbers even for fields that have the same semantical meaning as in pprof would diverge. With that in mind having field number gaps to allow pprof to add its new fields without breaking the representation compatibility seems like not achievable but maybe I'm missing something.